pupil_src/shared_modules/data_api/data_api_plugin.py

```diff
@@ -0,0 +1,95 @@
+"""
+(*)~---------------------------------------------------------------------------
+Pupil - eye tracking platform
+Copyright (C) 2012-2020 Pupil Labs
+
+Distributed under the terms of the GNU
+Lesser General Public License (LGPL v3.0).
+See COPYING and COPYING.LESSER for license details.
+---------------------------------------------------------------------------~(*)
+"""
+import logging
+
+from plugin import Plugin
+from pyglui import ui
+
+from .model import FrameFormat
+from .controller import FramePublisherController
+from .controller import PupilRemoteController
+from .ui import FramePublisherMenu
+from .ui import PupilRemoteMenu
+
+
+logger = logging.getLogger(__name__)
+
+
+class DataApiPlugin(Plugin):
+    menu_label = "Data API"
+    icon_chr = chr(0xE307)
+    icon_font = "pupil_icons"
+    order = 0.01  # excecute first
+
+    def __init__(self, g_pool, **kwargs):
+        super().__init__(g_pool)
+        self.__frame_publisher = FramePublisherController(**kwargs)
+        self.__frame_publisher.add_observer("on_frame_publisher_did_stop", self.on_frame_publisher_did_stop)
+
+
+        self.__pupil_remote = PupilRemoteController(g_pool, **kwargs)
+
+        # UI
+        self.__frame_publisher_ui_menu = FramePublisherMenu(self.__frame_publisher)
+        self.__pupil_remote_ui_menu = PupilRemoteMenu()
```

Comment:
UI should not be initialized in `__init__()` if possible.

Reply:
the components themselves don't call any UI function and don't create any `pyglui` instances in their `__init__`.

do you still think that initializing these objects should move to `init_ui`?